### PR TITLE
plugins.id("com.louiscad.splitties")

### PR DESCRIPTION
**Pull request by jmfayard**
- `cd plugin`
- Open project in IntelliJ IDEA
- Create account on https://plugins.gradle.org/
- `./gradlew login` # login to the Gradle Plugin Portal
- `./gradlew publish` # publish the plugin locally to the build repo
- `./gradlew publishPlugins` # publish to the Gradle Plugins Portal

Last command should fail with: 

```
> Task :publishPlugins
Publishing plugin com.louiscad.splitties version 0.0.1
Thank you. Your new plugin com.louiscad.splitties has been submitted for approval by Gradle engineers. The request should be processed within the next 24 hours, at which point you will be contacted via email.
Publishing artifact build/libs/Splitties-Plugin-0.0.1.jar
Publishing artifact build/libs/Splitties-Plugin-0.0.1-sources.jar
Publishing artifact build/libs/Splitties-Plugin-0.0.1-javadoc.jar
Publishing artifact build/publications/pluginMaven/pom-default.xml
Activating plugin com.louiscad.splitties version 0.0.1
```

Ping Eric Wendelin from Gradle to get it approved.

### CODE REVIEW

File: plugin/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+repositories {
+    mavenCentral()
+    jcenter()
+}
+
+pluginBundle {
+    website = "https://github.com/LouisCAD/Splitties"
+    vcsUrl = "https://github.com/LouisCAD/Splitties"
+    tags = listOf("android")
```
Comment (LouisCAD) [inline plugin/build.gradle.kts]: If these are tags for discovery, I'd add "kotlin".

File: plugin/gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.5.1-all.zip
```
Comment (LouisCAD) [inline plugin/gradle/wrapper/gradle-wrapper.properties]: Why not reuse the project's existing gradle wrapper? I'd prefer to do the update once per release for the whole project.

**Comment by jmfayard** [comment]
Tell Eric I was the first to try to publish the plugins but that the plugins should be owned by you

File: plugin/src/test/kotlin/com/louiscad/splitties/HelloTest.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.louiscad.splitties
+
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.StringSpec
+
+class HelloTest: StringSpec({
```
Comment (LouisCAD) [inline plugin/src/test/kotlin/com/louiscad/splitties/HelloTest.kt]: Is this dummy test really needed?

File: plugin/settings.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+rootProject.name = "Splitties-Plugin"
```
Comment (LouisCAD) [inline plugin/settings.gradle.kts]: Can't it be made a sub-project of the existing root project? Or is there reasons to not do it?
I'd like to share the library version (located in root project's `gradle.properties`) with the `Splitties` object's constants.

**Comment by jmfayard** [comment]
Hello, all of what you say can be done.
But the first step is to get the approval from Gradle to be published in
the Gradle plugins portal.

**Comment by LouisCAD** [comment]
Please, can you reuse the existing gradle distribution in the root project and remove the `bat`, `shell` and `jar` files, so I don't have to check for their authenticity before running the build?

**Comment by LouisCAD** [comment]
Thank you!

**Pull request merged**